x-pack/solutions/security/plugins/security_solution/public/configurations/page/configuration_tabs.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiTab, EuiTabs } from '@elastic/eui';
+import { useHistory, useParams } from 'react-router-dom';
+import { CONFIGURATIONS_PATH } from '../../../common/constants';
+import * as i18n from '../translations';
+export enum ConfigurationTabs {
+  integrations = 'integrations',
+  basicRules = 'basic_rules',
+  knowledgeSources = 'knowledge_sources',
+}
+
+export const ConfigurationsTabs = React.memo(() => {
+  const history = useHistory();
+  const params: { tab: ConfigurationTabs } = useParams();
+
+  const tabs = useMemo(
+    () => ({
+      [ConfigurationTabs.integrations]: {
+        id: ConfigurationTabs.integrations,
+        name: i18n.INTEGRATIONS,
+        disabled: false,
+        href: `${CONFIGURATIONS_PATH}/${ConfigurationTabs.integrations}`,
+      },
+
+      [ConfigurationTabs.knowledgeSources]: {
+        id: ConfigurationTabs.knowledgeSources,
+        name: i18n.KNOWLEDGE_SOURCES,
+        disabled: false,
+        href: `${CONFIGURATIONS_PATH}/${ConfigurationTabs.knowledgeSources}`,
+      },
+      [ConfigurationTabs.basicRules]: {
+        id: ConfigurationTabs.basicRules,
+        name: i18n.BASIC_RULES,
+        disabled: false,
+        href: `${CONFIGURATIONS_PATH}/${ConfigurationTabs.basicRules}`,
+      },
+    }),
+    []
+  );
+
+  const tabsArray = useMemo(() => Object.values(tabs), [tabs]);
+
+  const initialPage = params.tab || ConfigurationTabs.integrations;
+  const [selectedTabId, setSelectedTabId] = useState(initialPage);
```

Comment:
Do we really need this state? Can't we use `params.tab` directly?

Reply:
We do not, used the `params.tab` directly 👍